### PR TITLE
Skip release 456 in compatibility checks

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

**Comment by wendigo** [comment]
Compatibility tests are still failing because 459 was partially released. Merging as there is nothing we can do at this stage.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteCompatibility.java [hunk 1]
```diff
@@ -81,8 +81,8 @@ private static List<TestedImage> testedTrinoDockerImages()
             ImmutableList.Builder<TestedImage> testedTrinoVersions = ImmutableList.builder();
             int testVersion = currentVersion - 1; // always test last release version
             for (int i = 0; i < NUMBER_OF_TESTED_VERSIONS; i++) {
-                if (testVersion == 404) {
```
Comment (ebyhr) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteCompatibility.java:84]: @wendigo Is there no possibility to run `404` version in the future?

Comment (losipiuk) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteCompatibility.java:84]: there is no 404 version

Comment (ebyhr) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteCompatibility.java:84]: Yeah, I know it. The questions was whether this loop will hit 404 version in the future or not. I see this logic is different from JDBC compatibility test that runs n-th versions / 7. Never mind. 

https://github.com/trinodb/trino/blob/master/testing/trino-test-jdbc-compatibility-old-driver/bin/run_tests.sh#L15

Comment (losipiuk) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteCompatibility.java:84]: Oh - yeah - I did not notice you are commenting on removal.

Seems like number of versions and step are hardcoded right now. So we are past 404 already.
I think it makes sense to clean that up given current logic prohibits stepping on 404 land-mine.

**Pull request merged**